### PR TITLE
Fix legacyjit CI testing

### PR DESCRIPTION
**Pull request by rartemev**

### CODE REVIEW

**Comment by rartemev** [comment]
@dotnet-bot help

**Comment by jashook** [comment]
@dotnet-bot test CI Please

File: netci.groovy [hunk 1]
```diff
@@ -433,7 +437,7 @@ def static addNonPRTriggers(def job, def branch, def isPR, def architecture, def
                     }
                 }
                 // For x86, only add per-commit jobs for Windows
-                else if (architecture == 'x86') {
```
Comment (jashook) [inline netci.groovy]: I think we are good not having pr1r2r legacy_backend runs per commit. If we really want r2r jobs for legacy_backend it is probably worth doing once a day or once a week. @RussKeldorph may have a different opinion.

File: netci.groovy [hunk 2]
```diff
@@ -479,7 +483,7 @@ def static addNonPRTriggers(def job, def branch, def isPR, def architecture, def
...
+                else if (architecture == 'x86' || architecture == 'x86lb') {
```
Comment (jashook) [inline netci.groovy]: I do not think we need/want gcstress

**Comment by rartemev** [comment]
@dotnet-bot test CI Please

File: netci.groovy (hunk 2 above)
Comment (RussKeldorph) [inline netci.groovy]: Right, don't need gcstress.  Maybe later.

**Comment by rartemev** [comment]
@dotnet-bot test CI Please

File: netci.groovy (hunk 1 above)
Comment (RussKeldorph) [inline netci.groovy]: Lets see if we can live without them for now.  R2R is important for perf testing, and perf is the main reason for legacy_backend, so we may have to reevaluate.

**Comment by RussKeldorph** [comment]
Should we merge this into the dev/unix_test_workflow branch to test it first?

**Comment by jashook** [comment]
It will not hurt to get more testing there. @rartemev could you reopen this pr into dev/unix_test_workflow?

**Comment by rartemev** [comment]
@dotnet-bot test CI Please

**Comment by rartemev** [comment]
@jashook Sure. https://github.com/dotnet/coreclr/pull/12254

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
See https://ci.dot.net/job/dotnet_coreclr/job/dev_unix_test_workflow/job/x86_lb_debug_windows_nt/5 for the "successful" run.

File: tests/legacyjit_x86_testenv.cmd [hunk 3]
```diff
@@ -0,0 +1,10 @@
+@REM -------------------------------------------------------------------------
+@REM 
+@REM  This script provides x86 Legacy JIT test environment settings
+@REM
+@REM -------------------------------------------------------------------------
+
+set COMPLUS_AltJit=*
+set COMPLUS_AltJitName=legacyjit.dll
```
Comment (BruceForstall) [inline tests/legacyjit_x86_testenv.cmd]: You should also add:
```
set COMPLUS_AltJitNgen=*
```

File: netci.groovy [hunk 4]
```diff
@@ -1115,6 +1119,7 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
                 'russellhadley',
                 'RussKeldorph',
                 'sandreenko',
+                'rartemev',
                 'sdmaclea',
```
Comment (BruceForstall) [inline netci.groovy]: IMO, this should be part of a separate PR, since it's unrelated

File: netci.groovy [hunk 5]
```diff
@@ -365,6 +365,9 @@ def static getJobName(def configuration, def architecture, def os, def scenario,
         case 'x86':
             baseName = architecture.toLowerCase() + '_' + configuration.toLowerCase() + '_' + os.toLowerCase()
             break
+        case 'x86lb':
+            baseName = 'x86_lb_' + configuration.toLowerCase() + '_' + os.toLowerCase()
+            break
```
Comment (BruceForstall) [inline netci.groovy]: why use 'x86_lb_' instead of 'x86lb'? If you used 'x86lb' you could just add `case 'x86lb':` next to `case 'x86'` and reuse that code.

File: netci.groovy [hunk 6]
```diff
@@ -1624,6 +1656,7 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
             switch (architecture) {
                 case 'x64':
                 case 'x86':
+                case 'x86lb':
                     if (architecture == 'x86' && os == 'Ubuntu') {
```
Comment (BruceForstall) [inline netci.groovy]: Doesn't this add legacyjit testing support for Linux? I don't think we want or need that, do we?

Comment (rartemev) [inline netci.groovy]: Is it Ok if I add something like below there
```
assert architecture != 'x86lb' || os == 'Windows_NT'
```

File: tests/legacyjit_x86_testenv.cmd [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+@REM 
+@REM  This script provides x86 Legacy JIT test environment settings
+@REM
+@REM -------------------------------------------------------------------------
+
+set COMPLUS_AltJit=*
+set COMPLUS_AltJitNgen=*
+set COMPLUS_AltJitName=legacyjit.dll
+set COMPLUS_NoGuiOnAssert=1
+set COMPLUS_AltJitAssertOnNYI=1
```
Comment (BruceForstall) [inline tests/legacyjit_x86_testenv.cmd]: GitHub tells me there is no newline at the end of this file. Will that be a problem?

File: netci.groovy (hunk 6 above)
Comment (BruceForstall) [inline netci.groovy]: not sure it that is needed.

It looks like the line this comment was attached to has disappeared or moved, so maybe you changed the code, or maybe it is no longer relevant.

File: tests/legacyjit_x86_testenv.cmd (hunk 7 above)
Comment (rartemev) [inline tests/legacyjit_x86_testenv.cmd]: Added new line

**Comment by rartemev** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test
